packages/shared/src/chain.ts

```diff
@@ -257,7 +257,7 @@ export function applyScriptCondition({
 }) {
   // compile all folders in app directory, exclude node_modules
   rule.include.add({
-    and: [context.rootPath, { not: NODE_MODULES_REGEX }],
```

Comment:
Webpack supports the RegExp type, and rspack also previously supported the RegExp type. Isn't it supported now?
<img width="775" alt="image" src="https://github.com/web-infra-dev/rsbuild/assets/22373761/0c7406b0-dc17-4f15-8620-2f0b1c95ed8e">

Reply:
see https://github.com/web-infra-dev/rspack/issues/5057

Reply:
we can revert this change~